terragraph_planner/common/rf/link_budget_calculator.py

```diff
@@ -360,32 +360,21 @@ def get_max_boresight_gain_from_pattern_file(
 
 def get_max_tx_power(
     tx_sector_params: SectorParams,
-    tx_radio_pattern_data: Optional[Union[AntennaPatternData, ScanPatternData]],
     max_eirp_dbm: Optional[float],
 ) -> float:
     """
     Maximum Tx power = min(Max EIRP - Tx Antenna Gain, user input Max Tx power)
     """
     max_tx_power = tx_sector_params.maximum_tx_power
     if max_eirp_dbm is not None:
-        if tx_radio_pattern_data is not None and is_scan_pattern_data(
-            tx_radio_pattern_data
-        ):
-            tx_gain = get_max_boresight_gain_from_pattern_file(
-                cast(ScanPatternData, tx_radio_pattern_data)
-            )
-        else:
-            tx_gain = extract_gain_from_radio_pattern(
```

Comment:
I wonder if we need to replace other `extract_gain_from_radio_pattern` with the new way to compute.

Reply:
I'm not quite sure I understand.

This change simply takes the maximum value from those functions and uses it here to restrict tx power if tx power + gain exceeds max eirp.

Reply:
`extract_gain_from_radio_pattern` is replaced by a single add operation here, but I found there's other usage of `extract_gain_from_radio_pattern`, so 
 I wondered if they should be replaced too. https://github.com/terragraph/terragraph-planner/blob/main/terragraph_planner/common/rf/link_budget_calculator.py#L315

Reply:
No. Max eirp is a regulatory limit. It states that the gain plus tx power must be less than the specified value. In this PR, I'm choosing the max gain possible - this ensures that gain plus tx power <= max eirp regardless of the actual gain value.

Alternatively, we could make it orientation dependent. Then, we'd compute the actual gain here using those functions. That might be a subsequent update that will be made.